stylix/hm/icon.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, pkgs, ... }:
+
+let
+	cfg = config.stylix.iconTheme;
+	inherit (config.stylix) polarity;
+in {
+	imports = [ ../icon.nix ];
+	config = lib.mkIf (config.stylix.enable && cfg.enable && pkgs.stdenv.hostPlatform.isLinux) {
```

Comment:
Could you clarify why the `pkgs.stdenv.hostPlatform.isLinux` guard is used? If it is actually necessary, consider explaining it in a code comment just above.

Reply:
I believe some people use nix configs on macos, and I don't think you can change icons on mac in this way. I don't have mac to test it out so just for safety I'm restricting this option to only work on Linux, so people could easily use the same config across both devices without modified it on each device.

Reply:
> I believe some people use nix configs on macos, and I don't think you can change icons on mac in this way. I don't have mac to test it out

Me neither.

> just for safety I'm restricting this option to only work on Linux, so people could easily use the same config across both devices without modified it on each device.

I suggest optimistically removing the guard in this PR and re-adding the guard if someone complains in the future.

Reply:
Alright I can do it right now give me 5 minutes.